python/pyspark/util.py

```diff
@@ -55,7 +55,9 @@ def _get_argspec(f):
     """
     # `getargspec` is deprecated since python3.0 (incompatible with function annotations).
     # See SPARK-23569.
-    if sys.version_info[0] < 3:
+    if hasattr(f, '_argspec'):
```

Comment:
Can you add a comment explaining this? Just from this function it's not clear to me why do we need
`_argspec`

Reply:
Yea, the current way sounds a hack .. let's document this although we have a plan to clean this up soon.

Reply:
Sounds good